File: cblmapper/src/main/java/io/sportner/cblmapper/exceptions/CBLMapperClassException.java
package io.sportner.cblmapper.exceptions;

public class CBLMapperClassException extends RuntimeException {
    Class mUnhandledClass;
    String mMessage;

    public CBLMapperClassException(Class unhandledClass) {
        this.mUnhandledClass = unhandledClass;
    }

    public CBLMapperClassException(Class unhandledClass, String message) {
        this(unhandledClass);
        this.mMessage = message;
    }

    public String toString() {
        return String.format("CBLMapperClassException for class '%s': %s", this.mUnhandledClass, this.mMessage);
    }

    public Class getExceptionClass() {
        return this.mUnhandledClass;
    }
}

File: cblmapper/src/main/java/io/sportner/cblmapper/exceptions/CBLMapperClassException.kt
package io.sportner.cblmapper.exceptions

import java.lang.reflect.Type

open class CBLMapperClassException(protected val exceptionClass: Type,
                                   protected val description: String) : RuntimeException() {

    override fun toString(): String {
        return String.format("CBLMapperClassException for class '%s': %s", this.exceptionClass.toString(), this.description)
    }
}
